blob/drivertest/drivertest.go

```diff
@@ -465,6 +468,89 @@ func testCanceledWrite(t *testing.T, newHarness HarnessMaker, pathToTestdata str
 	}
 }
 
+// testMetadata tests writing and reading the key/value metadata for a blob.
+func testMetadata(t *testing.T, newHarness HarnessMaker) {
+	const key = "blob-for-metadata"
+	hello := []byte("hello")
+
+	tests := []struct {
+		name     string
+		metadata map[string]string
+		content  []byte
+		want     map[string]string
+		wantErr  bool
+	}{
+		{
+			name:     "empty",
+			content:  hello,
+			metadata: map[string]string{},
+			want:     nil,
+		},
+		{
+			name:     "empty key fails",
+			content:  hello,
+			metadata: map[string]string{"": "empty key value"},
+			wantErr:  true,
+		},
+		{
+			name:    "valid metadata",
+			content: hello,
+			metadata: map[string]string{
+				"key-a": "value-a",
+				"kEy-B": "value-b",
+				"key-c": "vAlUe-c",
+			},
+			want: map[string]string{
+				"key-a": "value-a",
+				"key-b": "value-b",
+				"key-c": "vAlUe-c",
+			},
+		},
```

Comment:
Test for two keys that differ only by case. I would expect an error.

Reply:
Done.